app/controllers/events_controller.rb

```diff
@@ -0,0 +1,37 @@
+class EventsController < ApplicationController
+  before_action :logged_in_user, only: [:new, :create]
+  def new
+    @event = Event.new
+  end
+
+  def create
+    unless current_user.try(:superadmin?)
```

Comment:
This can be written in a private method that checks if user is super admin. it will help you in solving the issue you had in https://github.com/AgileVentures/LocalSupport/pull/431#issuecomment-287629509

1. We can then have a before action to check if user is superuser 
``` ruby
class EventsController < ApplicationController
  before_action :is_superadmin?, except[:show, :index]
# rest of the class........
```
2. and the method is reduce to this
``` ruby
def create
@event = Event.new(event_params)
  if @event.save
    redirect_to @event, notice: 'Event was successfully created'
    else
      render :new
   end
end
```
3. the private method is then defined as
``` ruby
private
# previous code is here
def is_superadmin?
  unless current_user.try(:superadmin?)
  flash[:notice] = PERMISSION_DENIED
  redirect_to events_path and return false
end